src/lib/logical_query_plan/lqp_translator.cpp

```diff
@@ -146,65 +159,62 @@ std::shared_ptr<AbstractOperator> LQPTranslator::_translate_join_node(
   const auto input_operator = translate_node(node->left_child());
 
   const auto aggregate_node = std::dynamic_pointer_cast<AggregateNode>(node);
-  auto aggregate_expressions = aggregate_node->aggregate_expressions();
-  auto groupby_columns = aggregate_node->groupby_column_ids();
+  auto aggregate_expressions = _translate_expressions(aggregate_node->aggregate_expressions(), node);
+
+  std::vector<ColumnID> groupby_columns;
+  for (const auto& groupby_column_origin : aggregate_node->groupby_column_origins()) {
+    groupby_columns.emplace_back(node->left_child()->get_output_column_id_by_column_origin(groupby_column_origin));
+  }
 
   auto aggregate_input_operator = input_operator;
 
   /**
-   * 1. Handle arithmetic expressions in aggregate functions via Projection.
-   *
-   * In Hyrise, only Projections are supposed to be able to handle arithmetic expressions.
-   * Therefore, if we encounter an expression within an aggregate function, we have to execute a Projection first.
-   * The Aggregate will work with the output columns of that Projection.
+   * 1. Handle arithmetic expressions in the arguments of aggregate functions via a Projection.
    *
-   * We only need a Projection before the aggregate if the function argument is an arithmetic expression.
+   * If there are arithmetic expressions in aggregates, e.g. `a+b` in `SUM(a+b)`, we create a Projection to compute it.
+   * If there are no arithmetics expressions we don't need the Projection (i.e. need_projection is false)
    */
-  auto need_projection = false;
-
-  // Check if there are any arithmetic expressions.
-  for (const auto& aggregate_expression : aggregate_expressions) {
-    DebugAssert(aggregate_expression->type() == ExpressionType::Function, "Expression is not a function.");
-
-    // Check whether the expression that is the argument of the aggregate function, e.g. `a+b` in `SUM(a+b)` is, as in
-    // this case, an arithmetic expression and therefore needs a Projection before the aggregate is performed. If all
-    // Aggregates are in the style of `COUNT(b)` or there are just groupby columns, there is no need for a Projection.
-    const auto& function_arg_expr = (aggregate_expression->expression_list())[0];
-    if (function_arg_expr->is_arithmetic_operator()) {
-      need_projection = true;
-      break;
-    }
-  }
+  auto need_projection =
+      std::any_of(aggregate_expressions.begin(), aggregate_expressions.end(), [&](const auto& aggregate_expression) {
+        DebugAssert(aggregate_expression->type() == ExpressionType::Function, "Expression is not a function.");
+        return aggregate_expression->aggregate_function_arguments()[0]->is_arithmetic_operator();
+      });
 
   /**
-   * If there are arithmetic expressions create a Projection with:
-   *
-   *  - arithmetic expressions,
-   *  - columns used in aggregate functions, and
-   *  - GROUP BY columns
+   * If there are arithmetic expressions create a Projection with:   *
```

Comment:
stray `*`

Reply:
Also the "then" part is missing.

Reply:
removed the *. What "then" part. The comment makes sense to me

Reply:
Fresh day, fresh mind. Nevermind.